src/main/java/io/appium/java_client/remote/AppiumProtocolHandShake.java

```diff
@@ -54,12 +56,13 @@
 class AppiumProtocolHandShake {
 
     public Result createSession(HttpClient client, Command command)
-            throws IOException {
+            throws IOException, WebDriverException {
+
+        Capabilities desired = ofNullable((Capabilities) command.getParameters().get("desiredCapabilities"))
+                .orElse(new DesiredCapabilities());
```

Comment:
this will also create an extra instance of _DesiredCapabilities_. One must use _orElseGet_ here to avoid this (_orElse_ is good for constants only).